### PR TITLE
Adapt schema matching visualization to use the new api

### PR DESCRIPTION
**Pull request by EdenWuyifan**

### CODE REVIEW

**Review by aecio — CHANGES_REQUESTED** [review]
I tested it and the visualization is not running properly here. I'm going to commit some changes to fix some basic typing errors, but even after fixing them I still see errors like the one below. I also left some inline comments.

```
Select 'Histologic_Grade_FIGO' from the dropdown menu and observe the error message below.:

2024-07-08 21:46:40,388 ERROR: panel.reactive - Callback failed for object named "Column" changing property {'value': 'Histologic_Grade_FIGO'} 
Traceback (most recent call last):
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/reactive.py", line 383, in _process_events
    self.param.update(**self_events)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2318, in update
    restore = dict(self_._update(arg, **kwargs))
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2351, in _update
    self_._batch_call_watchers()
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2545, in _batch_call_watchers
    self_._execute_watcher(watcher, events)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2505, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/param.py", line 869, in _replace_pane
    self._update_inner(new_object)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/pane/base.py", line 704, in _update_inner
    new_pane, internal = self._update_from_object(
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/pane/base.py", line 678, in _update_from_object
    cls._recursive_update(old, new)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/pane/base.py", line 619, in _recursive_update
    cls._recursive_update(sub_old, sub_new)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/pane/base.py", line 644, in _recursive_update
    old.param.update(**new_params)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2318, in update
    restore = dict(self_._update(arg, **kwargs))
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2351, in _update
    self_._batch_call_watchers()
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2545, in _batch_call_watchers
    self_._execute_watcher(watcher, events)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2505, in _execute_watcher
    watcher.fn(*args, **kwargs)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/panel/pane/vega.py", line 198, in _update_selections
    self.selection.param.update({p: None for p in params})
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2318, in update
    restore = dict(self_._update(arg, **kwargs))
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 2344, in _update
    setattr(self_or_cls, k, v)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 528, in _f
    instance_param.__set__(obj, val)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 530, in _f
    return f(self, obj, val)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameterized.py", line 1497, in __set__
    self._validate(val)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameters.py", line 2520, in _validate
    self._validate_value(val, self.allow_None)
  File "/home/aeciosantos/workspace/askem-arpa-h/bdi-kit/venv/lib/python3.8/site-packages/param/parameters.py", line 2553, in _validate_value
    raise ValueError(
ValueError: List parameter 'Selection.single' must be a list, not an object of <class 'NoneType'>.
```

File: bdikit/visualization/schema_matching.py [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+        )
+        if exists(cache_path):
+            with open(cache_path) as f:
+                return json.load(f)
+        return None
+
+    def get_recommendations(
+        self, source: pd.DataFrame, target: pd.DataFrame, top_k: int
+    ) -> List[TopkMatching]:  # type: ignore
+        recommandations = []
```
[the remark this answers is not in the document]
Comment (aecio) [reply bdikit/visualization/schema_matching.py]: Fixed in commit 924106c.

File: bdikit/visualization/schema_matching.py [hunk 2]
```diff
@@ -0,0 +1,870 @@
...
+            f"reducings_{self._get_ground_truth_checksum()}_{self._get_data_checksum()}_{self.top_k}.json",
+        )
+        if exists(cache_path):
+            with open(cache_path) as f:
+                return json.load(f)
+        return None
+
+    def get_recommendations(
+        self, source: pd.DataFrame, target: pd.DataFrame, top_k: int
+    ) -> List[TopkMatching]:  # type: ignore
```
[the remark this answers is not in the document]
Comment (aecio) [reply bdikit/visualization/schema_matching.py]: Fixed in commit https://github.com/VIDA-NYU/bdi-kit/commit/924106cf169ddab9f5901ae04ce70236659c82aa.

File: bdikit/visualization/schema_matching.py [hunk 3]
```diff
@@ -0,0 +1,870 @@
...
+            **Values{is_sample}:** {sample}
+
+        """
+        else:
+            rec_info = f"""
+            ### Selected Recommendation
+
+            **Name:** {rec}
+
+            **Similarity:** {selected_row["Value"].values[0]}
```
[the remark this answers is not in the document]
Comment (EdenWuyifan) [reply bdikit/visualization/schema_matching.py]: Fixed

File: bdikit/visualization/schema_matching.py [hunk 4]
```diff
@@ -0,0 +1,870 @@
...
+            theme_classes=["thead-dark", "table-sm"],
+        )
+        return table_candidates
+
+    def _plot_column_histogram(
+        self, column: str, dataset: pd.DataFrame = None
+    ) -> "pn.pane.Markdown | alt.Chart":
+        if dataset is None:
+            dataset = self.source
+        if dataset[column].dtype == "float64":
```
[the remark this answers is not in the document]
Comment (EdenWuyifan) [reply bdikit/visualization/schema_matching.py]: Resolved

**Comment by aecio** [comment]
Looks like the error above may be related to the panel version. The latest panel version that supports Python 3.8 (which I had installed) is 1.2.3. So we either find a way to make this work with panel 1.2.3 or we might need to drop support for Python 3.8.

**Comment by roquelopez** [comment]
Yeah, I faced some issues related to the panel version (1.2.3). After changing the column in the combo box, it doesn't update correctly the section "Value Matches"  `pannel==1.4.4` fixed that problem.

**Comment by aecio** [comment]
Another issue I found: it looks like the right-side histogram is not shown after we accept the match. For example, select `BMI` and `bmi` from the Dou and Cao datasets, and click the button to accept the match. The right-side histogram will disappear.

**Comment by EdenWuyifan** [comment]
> Another issue I found: it looks like the right-side histogram is not shown after we accept the match. For example, select `BMI` and `bmi` from the Dou and Cao datasets, and click the button to accept the match. The right-side histogram will disappear.

It is because the selection is not obtained after the heatmap gets refreshed. I need to figure out a way to solve that in the later version. But I think we can merge the current version, this is a minor bug.

**Comment by EdenWuyifan** [comment]
> Yeah, I faced some issues related to the panel version (1.2.3). After changing the column in the combo box, it doesn't update correctly the section "Value Matches", `pannel==1.4.4` fixed that problem.

Regarding the panel version issue, panel 1.2.3 might have some bugs. However, my virtual environment with Python 3.8.14 and panel 1.2.3 works fine. We need to investigate this version issue further in the long run.

If the interaction part isn't working, one possible problem could be related to the browser cache. Try clearing the cache and refreshing Jupyter Lab/Notebook.

**Review by aecio — APPROVED** [review]

**Pull request merged**